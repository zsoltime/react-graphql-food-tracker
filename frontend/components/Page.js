import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';

import Header from './Header';
import Meta from './Meta';
import GlobalStyles from './GlobalStyles';
import { Content, StyledPage } from './Page.styles';
import defaultTheme from '../themes';

const Page = ({ children }) => (
  <ThemeProvider theme={defaultTheme}>
    <>
      <StyledPage>
        <Meta />
        <Header />
        <Content>{children}</Content>
      </StyledPage>
      <GlobalStyles />
    </>
  </ThemeProvider>
);

Page.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Page;
