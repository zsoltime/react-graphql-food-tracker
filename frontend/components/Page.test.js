import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Page from './Page';

describe('<Page />', () => {
  it('matches snapshot', () => {
    const ChildComponent = () => <p>Test</p>;
    const wrapper = shallow(
      <Page>
        <ChildComponent />
      </Page>
    );

    expect(
      toJSON(wrapper.find('Pagestyles__StyledPage'))
    ).toMatchSnapshot();
  });
});
