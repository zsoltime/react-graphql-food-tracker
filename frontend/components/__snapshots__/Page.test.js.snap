// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> matches snapshot 1`] = `
<Pagestyles__StyledPage>
  <Meta />
  <Header />
  <Pagestyles__Content>
    <ChildComponent />
  </Pagestyles__Content>
</Pagestyles__StyledPage>
`;
